lib/logstash/outputs/google_cloud_storage.rb

```diff
@@ -199,6 +197,18 @@ def close
 
   private
 
+  def legacy_encode(event, encoded)
+    # use-codec uses whatever the codec output exactly
+    return encoded if @output_format == 'use-codec'
+
+    # Deprecated logic
+    if @output_format == 'json'
+      LogStash::Json.dump(event.to_hash) + "\n"
```

Comment:
minor uptimization: I suggest setting a const such as
```ruby
NL = "\n".freeze
```
and use that instead of `"\n"`